### PR TITLE
[BEAM-11205] Google Libraries BOM 24.0.0 and associated deps

### PR DESCRIPTION
**Pull request by suztomo**
Upgrading Google Cloud Libraries BOM to the latest version 24.0.0 and associated dependencies to the compatible versions.

CC: @kennknowles 

# Linkage Check

No new linkage errors: https://gist.github.com/suztomo/37ffe6928c1f6ca6d6a063690d0823d6

# Change to ApiSurface.java for Guava's ParametricNullness annotation

Guava 31 introduced [ParametricNullness annotation](https://github.com/google/guava/commit/40faa2cb6a7e4aa9535b1a2a21265d533aca3eae) added to some of the Guava classes. As per the Javadoc, the annotation is targeted for Kotlin platform. (The annotation indicates the return value of generic function is not nullable iff the type in the type parameter is not nullable.)

Before the change in ApiSurface.java, the annotation was marked as unexpectedly exposed type.

` ./gradlew -p sdks/java/io/google-cloud-platform test --tests=GcpApiSurfaceTest` fails:

```
java.lang.AssertionError: 
Expected: API surface to include only:
	Classes in package "com.google.api.core"
	Classes in package "com.google.api.client.googleapis"
	Classes in package "com.google.api.client.http"
...
but: The following disallowed classes appeared on the API surface:
	interface com.google.common.base.ParametricNullness exposed via:
		interface com.google.common.base.ParametricNullness
		interface com.google.common.base.Function
		class com.google.common.base.Optional
		class com.google.cloud.bigtable.config.CallOptionsConfig
		class com.google.cloud.bigtable.config.BigtableOptions
		class org.apache.beam.sdk.io.gcp.bigtable.BigtableIO$Write
		class org.apache.beam.sdk.io.gcp.bigtable.BigtableIO
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.apache.beam.sdk.io.gcp.GcpApiSurfaceTest.testGcpApiSurface(GcpApiSurfaceTest.java:128)
```

https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/19434/testReport/

Example usage of the annotation:

```
  @ParametricNullness
  T get();
```

This means that the method `get()` is nullable if type parameter `T` is nullable. It's non-null if `T` is non-null.

This ParametricNullness annotation is not intend to be used other than Guava because it's not public. It's only visible to com.google.common.base package. ApiSurfaceTest should not detect this as exposed APIs.

Initially, I tried to add an exception for the annotation class in GcpApiSurfaceTest.java. However, because the annotation class is only visible in Guava's package, I couldn't add it. Instead, I went for modifying the logic of ApiSurface.java not to report non-public annotation classes. This is because non-public annotations are not visible to Beam users and thus we don't have to consider them as exposed API surfaces.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

`ValidatesRunner` compliance status (on master branch)
--------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Lang</th>
      <th>ULR</th>
      <th>Dataflow</th>
      <th>Flink</th>
      <th>Samza</th>
      <th>Spark</th>
      <th>Twister2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Go</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/">
          Build Status
        </a>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
    <tr>
      <td>Java</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_ULR/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Streaming/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2_Streaming/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
    <tr>
      <td>Python</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
    <tr>
      <td>XLang</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Samza/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
    </tr>
  </tbody>
</table>

Examples testing status on various runners
--------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Lang</th>
      <th>ULR</th>
      <th>Dataflow</th>
      <th>Flink</th>
      <th>Samza</th>
      <th>Spark</th>
      <th>Twister2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Go</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>Java</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Examples_Dataflow_Java11_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java_Examples_Dataflow_V2/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>Python</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
    <tr>
      <td>XLang</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
  </tbody>
</table>

Post-Commit SDK/Transform Integration Tests Status (on master branch)
------------------------------------------------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>Go</th>
      <th>Java</th>
      <th>Python</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
  </tbody>
</table>

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

<table>
  <thead>
    <tr>
      <th>---</th>
      <th>Java</th>
      <th>Python</th>
      <th>Go</th>
      <th>Website</th>
      <th>Whitespace</th>
      <th>Typescript</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Non-portable</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/">
          Build Status
        </a><br>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
    </tr>
    <tr>
      <td>Portable</td>
      <td>---</td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>
        <a href="https://ci-beam.apache.org/job/beam_PreCommit_GoPortable_Cron/lastCompletedBuild/">
          Build Status
        </a>
      </td>
      <td>---</td>
      <td>---</td>
      <td>---</td>
    </tr>
  </tbody>
</table>

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by suztomo** [comment]
Apache Spark Runner ValidatesRunner Tests 

```
10:28:21 > Task :runners:spark:2:validatesRunnerBatch
10:28:21 
10:28:21 org.apache.beam.sdk.transforms.ParDoLifecycleTest > testTeardownCalledAfterExceptionInFinishBundle FAILED
10:28:21     java.lang.AssertionError at ParDoLifecycleTest.java:300
```

https://ci-beam.apache.org/job/beam_PostCommit_Java_DataflowV2_PR/68/
```
10:56:14 > Task :runners:google-cloud-dataflow-java:googleCloudPlatformRunnerV2IntegrationTest
11:18:09 
11:18:09 org.apache.beam.sdk.io.gcp.pubsublite.ReadWriteIT > testReadWrite FAILED
11:18:09     java.lang.AssertionError at ReadWriteIT.java:258
```

https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/19440/testReport/junit/org.apache.beam.sdk.fn.data/BeamFnDataInboundObserver2Test/testCloseVisibleToAwaitCompletionCallerAndProducer/

```
Error Message
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Inbound observer closed.
Stacktrace
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Inbound observer closed.
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.apache.beam.sdk.fn.data.BeamFnDataInboundObserver2Test.testCloseVisibleToAwaitCompletionCallerAndProducer(BeamFnDataInboundObserver2Test.java:171)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Na

```

Retrying.

**Comment by suztomo** [comment]
Run Spark ValidatesRunner

**Comment by suztomo** [comment]
Run PostCommit_Java_DataflowV2

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Java precommit: https://ci-beam.apache.org/job/beam_PreCommit_Java_Commit/19450/console

```
18:27:02 > Task :sdks:java:io:common:buildDependents
18:27:02 Build timed out (after 120 minutes). Marking the build as aborted.
19:38:18 Build was aborted
19:38:18 Recording test results
19:38:18 [Checks API] No suitable checks publisher found.
19:38:22 [Error Prone] Sleeping for 5 seconds due to JENKINS-32191...
19:38:22 > Task :runners:java-fn-execution:test FAILED
19:38:23 
19:38:23 FAILURE: Build completed with 2 failures.
19:38:23 
19:38:23 1: Task failed with an exception.
19:38:23 -----------
19:38:23 * What went wrong:
19:38:23 Execution failed for task ':release:go-licenses:java:dockerRun'.
```

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by suztomo** [comment]
Run Portable_Python PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run PostCommit_Java_Dataflow

**Comment by suztomo** [comment]
Run PostCommit_Java_Hadoop_Versions

**Comment by suztomo** [comment]
Run Java PostCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
@kennknowles @aaltay 
36 successful checks!

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java [hunk 1]
```diff
@@ -712,11 +712,18 @@ private void addExposedTypes(Class<?> clazz, Class<?> cause) {
   private void addExposedTypes(Invokable<?, ?> invokable, Class<?> cause) {
     addExposedTypes(invokable.getReturnType(), cause);
     for (Annotation annotation : invokable.getAnnotations()) {
+      Class<? extends Annotation> annotationClass = annotation.annotationType();
+      if (!Modifier.isPublic(annotationClass.getModifiers())) {
+        // Non-public annotations are invisible to Beam users. Example: Guava's
+        // ParametricNullness is package-private and exists to support Kotlin platform
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java:718]: Interesting! Definitely relevant beyond Kotlin (for example https://checkerframework.org/releases/0.9.7/doc/checkers/nullness/quals/PolyNull.html) but your change is more general and obviously correct.

**Review by kennknowles — APPROVED** [review]

**Comment by suztomo** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by suztomo** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
32 successful checks!

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
Run Java PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java (hunk 1 above)
Comment (suztomo) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java]: Thank you for reference. I updated the comment:

> This annotation helps static analyzers including Kotlin platform.

**Comment by suztomo** [comment]
32 successful checks.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -443,27 +443,27 @@ class BeamModulePlugin implements Plugin<Project> {
...
+    def autovalue_version = "1.8.2"
     def aws_java_sdk_version = "1.11.974"
     def aws_java_sdk2_version = "2.15.31"
     def cassandra_driver_version = "3.10.2"
     def checkerframework_version = "3.10.0"
     def classgraph_version = "4.8.104"
     def errorprone_version = "2.3.4"
     def google_clients_version = "1.32.1"
-    def google_cloud_bigdataoss_version = "2.2.2"
+    def google_cloud_bigdataoss_version = "2.2.3"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:454]: Memo: this is not part of the Libraries BOM. So we don't always touch this when we upgrade the Libraries BOM.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 3]
```diff
@@ -543,13 +543,13 @@ class BeamModulePlugin implements Plugin<Project> {
...
+        google_api_services_cloudresourcemanager    : "com.google.apis:google-api-services-cloudresourcemanager:v1-rev20211017-$google_clients_version",
         google_api_services_dataflow                : "com.google.apis:google-api-services-dataflow:v1b3-rev20210818-$google_clients_version",
-        google_api_services_healthcare              : "com.google.apis:google-api-services-healthcare:v1-rev20210806-$google_clients_version",
-        google_api_services_pubsub                  : "com.google.apis:google-api-services-pubsub:v1-rev20210809-$google_clients_version",
-        google_api_services_storage                 : "com.google.apis:google-api-services-storage:v1-rev20210127-$google_clients_version",
+        google_api_services_healthcare              : "com.google.apis:google-api-services-healthcare:v1-rev20211016-$google_clients_version",
+        google_api_services_pubsub                  : "com.google.apis:google-api-services-pubsub:v1-rev20211012-$google_clients_version",
+        google_api_services_storage                 : "com.google.apis:google-api-services-storage:v1-rev20211018-$google_clients_version",
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:552]: These google-api-services artifacts upgrades are not strictly necessary when we upgrade the Libraries BOM. I usually put whatever latest available.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 4]
```diff
@@ -443,27 +443,27 @@ class BeamModulePlugin implements Plugin<Project> {
...
-    def guava_version = "30.1.1-jre"
+    def grpc_version = "1.41.0"
+    def guava_version = "31.0.1-jre"
     def hadoop_version = "2.10.1"
     def hamcrest_version = "2.1"
     def influxdb_version = "2.19"
     def httpclient_version = "4.5.13"
     def httpcore_version = "4.4.14"
-    def jackson_version = "2.12.4"
+    def jackson_version = "2.13.0"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:466]: Memo: this is not part of the Libraries BOM. So we don't always touch this when we upgrade the Libraries BOM.

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 5]
```diff
@@ -443,27 +443,27 @@ class BeamModulePlugin implements Plugin<Project> {
...
     def checkerframework_version = "3.10.0"
     def classgraph_version = "4.8.104"
     def errorprone_version = "2.3.4"
     def google_clients_version = "1.32.1"
-    def google_cloud_bigdataoss_version = "2.2.2"
+    def google_cloud_bigdataoss_version = "2.2.3"
     def google_cloud_pubsublite_version = "1.2.0"
-    def google_code_gson_version = "2.8.6"
-    def google_oauth_clients_version = "1.31.0"
+    def google_code_gson_version = "2.8.8"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:456]: Memo: this is not part of the Libraries BOM. So we don't always touch this when we upgrade the Libraries BOM.

**Pull request merged**